sdk/keyvault/Azure.Security.KeyVault.Shared/tests/Azure.Security.KeyVault.Shared.Tests/TestExtensions.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using NUnit.Framework;
+
+namespace Azure.Security.KeyVault.Tests
+{
+    internal static class TestExtensions
+    {
+        public static TestCaseData ConditionalIgnore(this TestCaseData source, bool condition, string reason)
```

Comment:
Why not just use an if in test data generation method?

Reply:
The methods get ugly and harder to read/maintain. I already have more intended use for making a shared source test project anyway. Just figured this was enough of a proverbial straw to finally split the project like I should've in the first place when I added it.